rep-0149.rst

```diff
@@ -0,0 +1,1047 @@
+distinguishing them from existing tags using an attribute.
+Since the API in e.g. ``catkin_pkg`` is closely following the XML structure
+using an attribute would have made the API more complicated and potentially
+changed the sematic of code using the existing dependencies.
+In order to support dependency group existing code needs to be extended
+anyway, therefore exposing new members / tags is straight forward.
+
+The content of the group dependency tag is the name of the group.
+Leading and trailing whitespace is being ignore from the name and for
+consistency it is recommended to following the naming rules for packages.
+
+There are different criteria for when a group dependency is being satisfied:
+
+* either all members of the group need to be available
+* or at least one member of the group needs to be available
+
+Other criteria would be possible but since there was no imminent use case
+this REP only defines these two.
+
+ABI version attribute
+---------------------
+
+The early draft of REP 127 proposed an attribute (`abi_version`) to identify
+the package version which is considered "compatible".
+
+Since no consensus could be reached about specifying ABI compatibility
+in general (for packages as well as libraries) the proposed attribute was
+removed from the draft (see related discussions [13]_, [14]_, [15]_).
+
+Since then the number of packages in a ROS distributions has increased
+(e.g. Indigo has more than 2500 packages) as well as the number of platforms
+ROS provides binary packages for.
+Therefore the need to avoid unnecessary rebuilds has increased.
+It is also desired to be able to encourage more frequent releases if they
+don't require downstream packages to be rebuild.
+
+Dependencies across ROS 1 and ROS 2
+-----------------------------------
+
+.. note:: figure out if there is a way to support this use case without
+  significant drawbacks
+
+.. raw:: html
+
+  </font>
+
+
+Data Representation
+===================
+
+The package manifest is an XML file, with restricted syntax.
+
+The only top-level element allowed is ``<package>``.  Immediately
+subordinate to that are several required or optional elements, defined
+here.  No other tags are permitted directly under the ``<package>``
+element.
+
+.. raw:: html
+
+  <font color="green">
+
+<package format="3">
+--------------------
+
+The ``<package>`` tag is the unique top-level tag in a ``package.xml``
+file.  All other tags are nested under it.
+
+.. raw:: html
+
+  </font>
+
+Attributes
+''''''''''
+
+  ``format="NUMBER"``
+
+  Specifying the ``package.xml`` format being used.  If not set,
+  ``format="1"`` would be assumed, which is *not* the format described
+  here.  For this interface, you *must* specify ``format="3"``.  If
+  you are using a different format, please refer to the relevant
+  specification.  REP-0127 [1]_ described format one, REP-0140 [12]_ format two.
+
+  .. raw:: html
+
+    <font color="red">
+
+  ``abi_version="MAJOR.MINOR.PATCH"``
```

Comment:
Do we want to allow partial matches, aka "MAJOR.MINOR" would match "MAJOR.MINOR.*" which would allow declarations like semantic versioning? http://semver.org/

Reply:
I don't see a use case where that would be beneficial over specifying `MAJOR.MINOR.0`. Therefore I would say "no".